### PR TITLE
Feature/JS-5093: Publishing

### PR DESCRIPTION
**Pull request by ra3orblade**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/ts/interface/block/layout.ts [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
-	TableRows	 = 4,
-	TableColumns = 5,
-
-	Footer		 = 100,
+	Row				 = 0,
+	Column			 = 1,
+	Div				 = 2,
+	Header			 = 3,
+	TableRows		 = 4,
+	TableColumns	 = 5,
```
Comment (haron-IV) [inline src/ts/interface/block/layout.ts:9]: Why to format it like that?

File: src/ts/lib/action.ts [hunk 2]
```diff
@@ -810,6 +810,18 @@ class Action {
 		analytics.event('ThemeSet', { id });
 	};
 
+	publish (objectId: string) {
+		C.ObjectPublish(S.Common.space, objectId, (message: any) => {
+			if (message.error.code) {
+				return;
+			};
```
Comment (haron-IV) [inline src/ts/lib/action.ts:817]: you can just skip brackets and make in online.
`if (message.error.code) return`

File: src/ts/lib/api/response.ts [hunk 3]
```diff
@@ -192,6 +192,13 @@ export const ObjectShow = (response: Rpc.Object.Show.Response) => {
 	};
 };
 
+export const ObjectPublish = (response: Rpc.Object.Publish.Response) => {
+	return {
+		cid: response.getPublishcid(),
+		key: response.getPublishfilekey(),
+	};
+};
+
```
Comment (haron-IV) [inline src/ts/lib/api/response.ts:201]:
```ts
export const ObjectPublish = (response: Rpc.Object.Publish.Response) => ({
		cid: response.getPublishcid(),
		key: response.getPublishfilekey(),
})
```
I find it nicer

**Review by haron-IV — APPROVED** [review]

File: src/ts/lib/action.ts (hunk 2 above)
Comment (ra3orblade) [reply src/ts/lib/action.ts:817]: We do not write one-line if clauses.

**Pull request closed without merging**